lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -309,13 +312,13 @@ FailureOr<FusedMatch> getFusedBrgemmSequenceFromProducer(Operation *op) {
       continue;
     }
 
-    // Make sure this is a chain, ie. at least once in inputs and outputs
-    int numUses = std::count(user->getOperands().begin(),
-                             user->getOperands().end(), op->getResult(0));
     // At least one input and the last operand (output) is the same buffer
-    if (numUses < 2 ||
-        user->getOperands()[user->getOperands().size() - 1] != op->getResult(0))
+    if (user->getOperands()[user->getOperands().size() - 1] !=
+            op->getResult(0) &&
```

Comment:
Shouldn't this be OR?

Reply:
No it should be and because the result should match either the last or the penultimate operand.

Reply:
Alright, I see that now. Thanks.